src/main/java/com/box/sdk/BoxSignRequestSigner.java

```diff
@@ -0,0 +1,655 @@
+        JsonObject jsonObj = new JsonObject();
+        JsonUtils.addIfNotNull(jsonObj, "email", this.email);
+        JsonUtils.addIfNotNull(jsonObj, "name", this.name);
+        JsonUtils.addIfNotNull(jsonObj, "role", this.role);
+        JsonUtils.addIfNotNull(jsonObj, "is_in_person", this.isInPerson);
+        JsonUtils.addIfNotNull(jsonObj, "order", this.order);
+        JsonUtils.addIfNotNull(jsonObj, "language", this.language);
+        JsonUtils.addIfNotNull(jsonObj, "verification_phone_number", this.verificationPhoneNumber);
+        JsonUtils.addIfNotNull(jsonObj, "embed_url_external_user_id", this.embedUrlExternalUserId);
+        JsonUtils.addIfNotNull(jsonObj, "redirect_url", this.redirectUrl);
+        JsonUtils.addIfNotNull(jsonObj, "declined_redirect_url", this.declinedRedirectUrl);
+        JsonUtils.addIfNotNull(jsonObj, "verification_password", this.verificationPassword);
+
+        return jsonObj;
+    }
+
+    /**
+     * Type of decision made by signer.
+     */
+    public enum BoxSignRequestSignerDecisionType {
+
+        /**
+         * Signed decision.
+         */
+        Signed("signed"),
+
+        /**
+         * Declined decision.
+         */
+        Declined("declined");
+
+        private final String jsonValue;
+
+        private BoxSignRequestSignerDecisionType(String jsonValue) {
+            this.jsonValue = jsonValue;
+        }
+
+        static BoxSignRequestSignerDecisionType fromJSONString(String jsonValue) {
+            if (jsonValue.equals("signed")) {
+                return Signed;
+            } else if (jsonValue.equals("declined")) {
+                return Declined;
+            } else {
+                throw new IllegalArgumentException("The provided JSON value isn't a valid signed decision type.");
+            }
+        }
+    }
+
+    /**
+     * Represents a type of input.
+     */
+    public enum BoxSignRequestInputType {
+
+        /**
+         * Signature input.
+         */
+        Signature("signature"),
+
+        /**
+         * Text input.
+         */
+        Text("text"),
+
+        /**
+         * Checkbox input.
+         */
+        Checkbox("checkbox"),
+
+        /**
+         * Date input.
+         */
+        Date("date");
+
+        private final String jsonValue;
+
+        private BoxSignRequestInputType(String jsonValue) {
+            this.jsonValue = jsonValue;
+        }
+
+        static BoxSignRequestInputType fromJSONString(String jsonValue) {
+            if (jsonValue.equals("signature")) {
+                return Signature;
+            } else if (jsonValue.equals("text")) {
+                return Text;
+            } else if (jsonValue.equals("checkbox")) {
+                return Checkbox;
+            } else if (jsonValue.equals("date")) {
+                return Date;
+            } else {
+                throw new IllegalArgumentException("The provided JSON value isn't a valid signer input type.");
+            }
```

Comment:
switch

Reply:
Strings in switch are not supported in -source 1.6.